hack/build-release.sh

```diff
@@ -16,27 +16,49 @@
 
 set -eux
 
+[[ -n "${KO_DOCKER_REPO-}" ]] || ( echo KO_DOCKER_REPO must be set; exit 1)
```

Comment:
nit: mind adding a comment about the conventions here or turn it into a function? There's a lot of bash shorthand to unpack here.

Reply:
Functionized it